packages/remix/src/utils.ts

```diff
@@ -369,3 +390,14 @@ async function ensureSymlink(
   await fs.symlink(relativeTarget, symlinkPath);
   debug(`Created symlink for "${pkgName}"`);
 }
+
+export function isESM(path: string): boolean {
```

Comment:
How do I test this? If I make my `remix.config.js` file ESM and add `"type": "module"` in the `package.json`, `vc dev` blows up with the error:

```
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/chris/code/sites/chrisbarber-remix/build/index.js from /Users/chris/code/sites/chrisbarber-remix/node_modules/.pnpm/@remix-run+serve@1.19.1/node_modules/@remix-run/serve/dist/index.js not supported.
/Users/chris/code/sites/chrisbarber-remix/build/index.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename /Users/chris/code/sites/chrisbarber-remix/build/index.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/chris/code/sites/chrisbarber-remix/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

    at /Users/chris/code/sites/chrisbarber-remix/node_modules/.pnpm/@remix-run+serve@1.19.1/node_modules/@remix-run/serve/dist/index.js:43:17
<snip>
```

When I `vc build --prod && bc deploy --prebuilt --prod`, it deploys, but blows up at runtime with a 500:

```
file:///var/task/build/server-build-nodejs-eyJydW50aW1lIjoibm9kZWpzIn0.mjs:12
import build from './build-nodejs-eyJydW50aW1lIjoibm9kZWpzIn0.js';
       ^^^^^
SyntaxError: The requested module './build-nodejs-eyJydW50aW1lIjoibm9kZWpzIn0.js' does not provide an export named 'default'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:190:5)
RequestId: ac168cec-2088-4662-9595-755adef1b306 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

Reply:
Remix doesn't support `"type": "module"`. This bit of code is specifically for the `remix.config` file, in which `remix.config.mjs` is supported.